networklibrary/src/main/java/com/eis/smsnetwork/SMSNetworkManager.java

```diff
@@ -105,7 +162,9 @@ public void acceptJoinInvitation(Invitation invitation) {
      * @param list A NetSubscriberList of type <SMSPeer> to provide
      */
     public void setNetSubscriberList(@NonNull NetSubscriberList<SMSPeer> list) {
-
+        for (SMSPeer sub : netSubscribers.getSubscribers())
+            list.addSubscriber(sub);
+        netSubscribers = list;
```

Comment:
Wait what? This has an extra step of assignment, could have used `foreach(sub in list) netSub.addSub(sub)` and that's it.